internal/devbox/packages.go

```diff
@@ -43,6 +44,48 @@ const StateOutOfDateMessage = "Your devbox environment may be out of date. Run %
 // packages.go has functions for adding, removing and getting info about nix
 // packages
 
+type UpdateVersion struct {
+	Current string
+	Latest  string
+}
+
+// Outdated returns a map of package names to their available latest version.
+func (d *Devbox) Outdated(ctx context.Context) (map[string]UpdateVersion, error) {
+	ctx, task := trace.NewTask(ctx, "devboxOutdated")
+	defer task.End()
+
+	outdatedPackages := map[string]UpdateVersion{}
+
+	for _, pkg := range d.AllPackages() {
+		if strings.HasSuffix(pkg.Versioned(), "latest") {
+			continue
+		}
```

Comment:
`@latest` packages have their versions pinned in the devbox.lock file. I think you could get their current version via:

```
lockfile := d.Lockfile()
lockfile.Packages[p.Raw].Version
```

I think its useful to print if they're outdated as well. A user may want to know that.